examples/example_led.c

```diff
@@ -0,0 +1,136 @@
+#include <r502.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <limits.h>
+#include <errno.h>
+
+
+int main(int argc, char *argv[]) {
+    Driver *driver;
+    Reply reply;
+    int32_t err = SUCCESS;
+
+    if (argc == 2 && strncmp(argv[1], "--help", 6) == 0) {
+        printf("Usage: %s <port> <control> <speed> <color> <times>\n", argv[0]);
+        printf("port - your port, ex: \"/dev/ttyS0\" if you're using UART on RaspberryPi\n");
+        printf("control   1: breathing light\n");
+        printf("          2: flashing light\n");
+        printf("          3: light always on\n");
+        printf("          4: light always off\n");
+        printf("          5: light gradually on\n");
+        printf("          6: light gradually off\n");
+        printf("speed     0-255 It is effective for breathing lamp and flashing lamp,Light gradually on,Light gradually off\n");
+        printf("color     1: red\n");
+        printf("          2: blue\n");
+        printf("          3: purple\n");
+        printf("times     1-255, 0 for infinite\n");
+        return 0;
+    }
+
+    if (argc != 6) {
+        printf("Usage: %s <port> <control> <speed> <color> <times>\nUse %s --help for help\n", argv[0], argv[0]);
+        return -1;
+    }
+
+    char *controlPointer;
+    char *speedPointer;
+    char *colorPointer;
+    char *timesPointer;
+
+    int control;
+    int speed;
+    int color;
+    int times;
+
+    errno = 0;
+    long controlConv = strtol(argv[2], &controlPointer, 10);
+
+    // Check for errors: e.g., the string does not represent an integer
+    // or the integer is larger than int
+    if (errno != 0 || *controlPointer != '\0' || controlConv > INT_MAX || controlConv < INT_MIN) {
```

Comment:
This checks can be merged with the ones validating the arguments value ([here](https://github.com/DanBrezeanu/R502-fingerprint/pull/3/files#diff-1c8c3d17bded11a5c6ac29c701f146226ffd7ad394d5549257d65e8b60eb6565R89) is the matching check) so checking boundaries against INT_MIN/MAX can be avoided. 

Just be sure to check for `errno` and `*controlPointer` first in the OR expression

Reply:
I've redone the whole checking section of the code